GentenStudios/Phoenix
# New and Improved texture handler.

Resolves: #303
Authors:
@beeperdeeper089 

## Summary of changes
- Replace old texturing system with a new modular one.
- Allowed multiple texture sizes to be done.
- Errors are handled a lot better - no random crashes if a texture isn't loaded right.
  
## Caveats
Does this introduce any new bugs?
- Unknown textures haven't been setup yet (will do in a later PR once I have a better way to put the unknown texture data straight into code)
- Issue with texture wrapping with some seams in places. This is mostly fixed with ``GL_CLAMP_TO_EDGE`` as the wrapping option and anisotropic filtering. This can be further improved by anti-aliasing but I'll implement that better using a renderbuffer later on rather than relying on SDL's inbuilt stuff since we can then have some post-processing stuff too :).

## On approval
Merge, wait for @tobyplowy to give this a final test.

Testing Notes:
    - Make sure all textures are loading correctly in the right place in the right blocks.
    - Make sure the slope works, I remember fixing it but needs some more solid testing.
    - @tobyplowy you know what to do :wink:
